storage/flux/reader.go

```diff
@@ -560,11 +616,81 @@ func (wai *windowAggregateIterator) Do(f func(flux.Table) error) error {
 	if rs == nil {
 		return nil
 	}
-	return wai.handleRead(f, rs)
+	return wai.handleRead(f, rs, aggregate)
 }
 
-func (wai *windowAggregateIterator) handleRead(f func(flux.Table) error, rs storage.ResultSet) error {
-	return nil
+func (wai *windowAggregateIterator) handleRead(f func(flux.Table) error, rs storage.ResultSet, aggregate bool) error {
+	// these resources must be closed if not nil on return
+	var (
+		cur   cursors.Cursor
+		table storageTable
+	)
+
+	defer func() {
+		if table != nil {
+			table.Close()
+		}
+		if cur != nil {
+			cur.Close()
+		}
+		rs.Close()
+		wai.cache.Release()
+	}()
+
+READ:
```

Comment:
Is this the idiomatic way to walk an iterator in go? I'll admit to having some twitchiness over GOTO, but if it's normally how this is done in go, it's fine.

Reply:
Actually there isn't a GOTO for this label. This label is used because there's a select statement inside of a for loop. By placing a label before the for loop, calling `break LABEL` inside of the select will also break out of the for loop.

This removes the need for multiple break statements and the additional logic that goes with them. It's a pretty common pattern in Go, from what I've observed.

Reply:
Ah yes, okay, this makes sense as a "this is how we get around multiple meanings of break". Thanks for humoring me.